### ISSUE TITLE
Add ability to auto-distribute to all connected sites (internal and external)

### ISSUE THREAD

**Issue by mlangone** [10up/distributor#456]
**Describe your question**

I have a large multisite and would like to distribute a post to all my subsites, Is there a way to do this without having to click select 350 times? I can add filters or hooks as needed, I wanted to ask before our dev team got into trying to "make" it work.

Thanks
ML

**Comment by jeffpaul** [comment]
@mlangone what would be the ideal way in which you avoid "having to click select 350 times"?  Is that a "select all" option?  Having posts just auto-distribute to all connected sites?  Or something else?  Depending on your ideal approach there's likely different options/recommendations for you.

**Comment by mlangone** [comment]
Best option would be auto-distribute! We have a unique setup that if we could take it a step further would be even mo betta. :)
Something like this would be great
Mainsite.com ->autodistribute -> 350 subsites
sub1.mainsite.com - 40 subsites . Different server and installation
sub2.mainsite.com -> 25 subsites . Second different installation.

Currently mainsite.com -> distributes to blog_1 of sub1.mainsite.com, then we have to distribute to the other sites. We would like to be able to distribute to the 40 subsites

Is there a filter that we can just enable?

ML

**Comment by jeffpaul** [comment]
@mlangone no, there's no filter that would cover those features.  What you're asking for sounds like we'd be getting into a heavily customized Distributor setup, not something that's supported by the plugin as-is.

If you have a more urgent need to add this sort of auto-distribution support, then we have paid implementation and support options where we could help resolve this for you.  If that's the approach you'd like to take then email me at jeff.paul@10up.com and we can craft a small engagement to help... thanks!

**Comment by coenve** [comment]
Hi jeffpaul,

Just a reminder that there is still a need for this feature, it's been a while since this was discussed. Maybe one to schedule in the roadmap?

**Comment by arsendovlatyan** [comment]
Hey there,

We have developed an add-on, to be able to distribute large amount of posts via external connections using background tasks, but it was on a complicated setup and currently it's not in "plug and play" state, anyway here is it if you would like to check  https://github.com/NovemBit/distributor-in-background.

From our experience (it was with external connections only) I can say that it's not an easy thing to push large amount of new posts using background tasks, especially error reporting is very important, otherwise you can find many duplicated posts in destination website.

I would really like to see this kind-of functionality in Distributor's core, ready to help as much as I can.

**Comment by jeffpaul** [comment]
@coenve we have a couple high-priority items targeted for the next two releases, so this item would be unlikely to make it into Distributor core before then.  If this is of particular urgent/important need for you, then we can discuss alternate options for you, just shoot me an email to chat (jeff.paul@10up.com)... thanks!

**Comment by Sukalo0290** [comment]
Hey,

is this feature still being worked on?

I'm trying to add a cron or something to automatically push custom posts types and posts out to all sites internal and external or be able to specify which sites get pushed to.

I have tried the dist in background with not much joy.

**Comment by jeffpaul** [comment]
@Sukalo0290 you might check some of the filters added in [Distributor 2.0.0-beta2](https://github.com/10up/distributor/releases/tag/2.0.0-beta2) to see if they are ones you could interrogate for your needs here?

**Comment by peterwilsoncc** [comment]
I forgot to link the pull request introducing Auto Distribution to this issue, see PR https://github.com/10up/distributor/pull/1321. The feature was released in [version 2.2.0](https://github.com/10up/distributor/releases/tag/2.2.0) a few days ago.

You can review the tutorial for [enabling the feature in the developer documentation](https://10up.github.io/distributor/tutorial-auto-distribution.html).

As Distribution is deferred to WP-Cron, it's probably best to be running real cron (either via WP-CLI, Cron Control or Cavalcade) for the use case mentioned of distributing to several hundred sites.

I'll close this ticket off as completed.
